xwiki-platform-core/xwiki-platform-user/xwiki-platform-user-profile/xwiki-platform-user-profile-ui/src/main/resources/XWiki/XWikiUserPreferencesSheet.xml

```diff
@@ -37,6 +37,148 @@
+        <prettyName>Parse content</prettyName>
+        <unmodifiable>0</unmodifiable>
+        <classType>com.xpn.xwiki.objects.classes.BooleanClass</classType>
+      </parse>
+      <use>
+        <cache>0</cache>
+        <disabled>0</disabled>
+        <displayType>select</displayType>
+        <freeText>forbidden</freeText>
+        <largeStorage>0</largeStorage>
+        <multiSelect>0</multiSelect>
+        <name>use</name>
+        <number>3</number>
+        <prettyName>Use this extension</prettyName>
+        <relationalStorage>0</relationalStorage>
+        <separator> </separator>
+        <separators>|, </separators>
+        <size>1</size>
+        <unmodifiable>0</unmodifiable>
+        <values>currentPage|onDemand|always</values>
+        <classType>com.xpn.xwiki.objects.classes.StaticListClass</classType>
+      </use>
+    </class>
+    <property>
+      <cache>forbid</cache>
+    </property>
+    <property>
+      <code>require(['jquery', 'xwiki-meta'], function ($, xm) {
+  /**
+   * Handle the shortcut preference modifications
+   */
+  $(document).off('input.shortcutPreference');
+  let updateShortcut = function( shortcutPreferenceDOM) {
+    let shortcutPreference = $(shortcutPreferenceDOM);
+    let translateKey = shortcutPreference.attr("data-translatekey");
+    let newValue = shortcutPreference.val();
+    let url = xm.restURL;
+    let oldItemNumber = shortcutPreference.attr("data-shortcutprefnumber");
+    if (oldItemNumber == null) {
+      // Create the new user preference
+      let creationUrl = url + "/objects";
+      let data = {
+        'className': 'XWiki.UserPreferenceShortcutsClass',
+        'property#translationKey': translateKey,
+        'property#value': newValue
+      };
+      let objectCreationPromise = $.post(creationUrl, data, function(newObject) {
+        shortcutPreference.attr("data-shortcutprefnumber", newObject.firstChild.getElementsByTagName("number")[0].textContent);
+      });
+    } else {
+      if (newValue == shortcutPreference.attr("data-defaultvalue")) {
+        // Remove the existing preference, we don't want to keep it if the user is back to normal.
+        let destructionUrl = url + "/objects/XWiki.UserPreferenceShortcutsClass/" + oldItemNumber;
+        $.ajax({url: destructionUrl,
+          type: "DELETE"
+        });
+        shortcutPreference.removeAttr("data-shortcutprefnumber");
+      } else {
+        // Update the existing preference
+        let updateUrl = url + "/objects/XWiki.UserPreferenceShortcutsClass/" + oldItemNumber +"/properties/value";
+        let data = {
+          'property#value': newValue
+        };
+        $.ajax({url: updateUrl,
+          type: "PUT",
+          data: data
+        });
+      }
+    }
+  };
+  var timeoutId = 0;
+  $(document).on('input.shortcutPreference', '.shortcutPreference', function () {
+    if (timeoutId != 0){clearTimeout(timeoutId)}
+    timeoutId = setTimeout(updateShortcut, 500, this);
+  });
```

Comment:
Isn't it a bit surprising to save during editing? Wouldn't it be better to have a dedicated save button?

Reply:
It was not questionned on the forum topic. As proposed by Thiago, we could improve this by having "inline editing" for those properties. I think it goes beyong the scope of this first PR and I'd rather bring improvements like this in a different PR.

--> https://github.com/xwiki/xwiki-platform/pull/2981#issuecomment-2396554277

Reply:
My point is mostly that these inputs are presented in a page that has explicit "Save" and "Cancel" buttons at the bottom - not applying these buttons to the keyboard shortcut preferences seems weird from a usability point of view. Anyway, I'm not going to insist on this if others think that this is okay.